src/Adapter/Image/ImageValidator.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Image;
+
+use ImageManager;
+use ImageManagerCore;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\ImageFileNotFoundException;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\ImageUploadException;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\MemoryLimitException;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\UploadedImageConstraintException;
+use Tools;
+
+/**
+ * Responsible for validating image before upload
+ */
+class ImageValidator
+{
+    /**
+     * @var int
+     */
+    private $maxUploadSize;
+
+    /**
+     * @param int|null $maxUploadSize
+     */
+    public function __construct(?int $maxUploadSize = null)
+    {
+        $this->maxUploadSize = $maxUploadSize ?: Tools::getMaxUploadSize();
+    }
+
+    /**
+     * @param string $filePath
+     *
+     * @throws ImageUploadException
+     * @throws UploadedImageConstraintException
+     */
+    public function assertFileUploadLimits(string $filePath): void
```

Comment:
I'm still wondering if the flow is correct.
**I Imagine this in http:**
1. User uploads a file via form
2. Form constraints are applied  (same validation actually - checks size and type in form).
3. move_uploaded_file to ps/tmp
4. Invoke `AddProductImageHandler` in which `ImageValidator` validates the file once again (it's the 2nd validation - the one that we are talking about right now).

**Other scenario using terminal:**
first 3 steps are skipped
4. Invoke `AddProductImageHandler` ... same result

Reply:
I think it's ok to **move** the file two times Actually that's why I don't like the term "upload" Because we don't upload anything here we just move files Files that were already uploaded, so it's a bit confusing

But I guess it's ok, the max size is probably redundant since it can be blocked on the request level by the server configuration It makes more sense in case we use the command from a terminal, except the file would already be uploaded so limiting its size won't bring much from a hard drive point of view 😅

But again I think for now it's fine this way